dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/config/ServiceConfig.java

```diff
@@ -77,7 +76,8 @@ public StyleManager styleManager(
 
   @Bean("org.hisp.dhis.outboundmessage.OutboundMessageService")
   public DefaultOutboundMessageBatchService defaultOutboundMessageBatchService(
-      SmsMessageSender smsMessageSender, EmailMessageSender emailMessageSender) {
+      @Qualifier("smsMessageSender") MessageSender smsMessageSender,
+      @Qualifier("emailMessageSender") MessageSender emailMessageSender) {
```

Comment:
I am confused. Didn't we discuss to use specific interfaces? This is like the opposite. Maybe I misunderstood?

Reply:
See first next step. I had to touch many files so the PR would not be as easy to review. I also could not get it to work. On the one hand most components either need email/sms but there are a couple that tie them together like `DefaultMessageService` and `DefaultOutboundMessageBatchService`. I need to analyze this again and figure out how we can make them replaceable in specific tests.